--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -168,6 +168,12 @@
 #include "devmem.h"
 #include "net-sysfs.h"
 
+#if defined(CONFIG_RTL8196E_IMEM) && defined(CONFIG_RTL8196E_IMEM_DEFAULT_PLACEMENT)
+#define __iram_hotpath __attribute__((section(".iram")))
+#else
+#define __iram_hotpath
+#endif
+
 static DEFINE_SPINLOCK(ptype_lock);
 struct list_head ptype_base[PTYPE_HASH_SIZE] __read_mostly;
 
@@ -6721,7 +6727,7 @@
  * The entry's receive function will be scheduled to run.
  * Consider using __napi_schedule_irqoff() if hard irqs are masked.
  */
-void __napi_schedule(struct napi_struct *n)
+void __iram_hotpath __napi_schedule(struct napi_struct *n)
 {
 	unsigned long flags;
 
@@ -6773,7 +6779,7 @@
  * because the interrupt disabled assumption might not be true
  * due to force-threaded interrupts and spinlock substitution.
  */
-void __napi_schedule_irqoff(struct napi_struct *n)
+void __iram_hotpath __napi_schedule_irqoff(struct napi_struct *n)
 {
 	if (!IS_ENABLED(CONFIG_PREEMPT_RT))
 		____napi_schedule(this_cpu_ptr(&softnet_data), n);
@@ -6782,7 +6788,7 @@
 }
 EXPORT_SYMBOL(__napi_schedule_irqoff);
 
-bool napi_complete_done(struct napi_struct *n, int work_done)
+bool __iram_hotpath napi_complete_done(struct napi_struct *n, int work_done)
 {
 	unsigned long flags, val, new, timeout = 0;
 	bool ret = true;
